src/Algebra/Graph.hs

```diff
@@ -395,6 +395,8 @@ foldg e v o c = go
 isSubgraphOf :: Ord a => Graph a -> Graph a -> Bool
 isSubgraphOf x y = overlay x y == y
 
+{-# SPECIALISE isSubgraphOf :: Graph Int -> Graph Int -> Bool #-}
```

Comment:
Can you skip the empty line above? It's not immediately obvious to which function the specialisations apply.

Reply:
Of course, I thought that rendered not so well, but you are right

Reply:
In fact, because it looks like a type definition, maybe adding it just below the actual type definition is a good idea

Reply:
Another option is to group all these pragmas together, say at the bottom of the file with a comment explaining what it achieves.

Reply:
Yes, but one can forget to check this. I like that when I am reading the code and I have everything under my eyes

Reply:
OK, let's place it next to function definitions. Perhaps, below the comment but on top of function type signature? Unless this messes up Haddock generation.

Reply:
Ok I will check this